src/ReadingBuddy.java

```diff
@@ -248,26 +250,122 @@ public static ArrayList<Integer> initialGetBookGroups(){
+            data.add(new String[] {"group_id","book_id","title","author","member1","member2","pages_read1","pages_read2","is_done1","is_done2"});
+            for(int x = 0; x < allBookGroups.size(); x++){
+                if(toRemove.getID() == allBookGroups.get(x).getID()){
+                    allBookGroups.remove(x);
+                } else {
+                    data.add(new String[] {Integer.toString(allBookGroups.get(x).getID()), Integer.toString(allBookGroups.get(x).getBookID()),
+                        allBookGroups.get(x).getTitle(), allBookGroups.get(x).getAuthor(), allBookGroups.get(x).getMembers().get(0), 
+                        allBookGroups.get(x).getMembers().get(1), Integer.toString(allBookGroups.get(x).getCurrentPages().get(0)),
+                        Integer.toString(allBookGroups.get(x).getCurrentPages().get(1)), Boolean.toString(allBookGroups.get(x).getIsDone().get(0)),
+                        Boolean.toString(allBookGroups.get(x).getIsDone().get(1))});
+                }
+            }
+            writer.writeAll(data); 
+            // closing writer connection 
+            writer.close(); 
+        } 
+        catch (IOException e) { 
+            e.printStackTrace(); 
+        } 
+    }
+
+    //NEW SPRINT3: updated to be able to use with currentUser's currentbook or their book buddy's
+    public static void removeCurrentBook(CurrentBook toRemove){
         //remove from data file
         File file = new File("reading_bubby/appdata/currently_reading.csv"); 
         try { 
             FileWriter outputfile = new FileWriter(file); 
             CSVWriter writer = new CSVWriter(outputfile);
             List<String[]> data = new ArrayList<String[]>();
-            data.add(new String[] {"book_id","title","author","username","current_page","current_chapter","group_id","book_buddy"});
+            data.add(new String[] {"book_id","title","author","username","current_page","current_chapter","group_id","book_buddy","has_open_search","is_done"});
             for(int x = 0; x < allCurrentBooks.size(); x++){
                 if((toRemove.getID() == allCurrentBooks.get(x).getID()) && (toRemove.getUsername().equals(allCurrentBooks.get(x).getUsername()))){
                     //when the selected book is found do not write it to the data but remove it from the user's ArrayList and allCurrentBooks Arraylist
-                    currentUser.removeFromCurrentlyReading(idxForUserArr);
+                    //NEW SPRINT 3 (misc--should delete search post if has one)
+                    if(toRemove.getHasOpenSearch()){
+                        SearchPost postToDel = searchForSpecificSearchPost(toRemove.getTitle(), toRemove.getAuthor(),
+                         toRemove.getUsername());
+                        //below if should not be necessary because the if currently in has checked if there is an associated search post
+                        if(postToDel.getBookID() == -1){
```

Comment:
it took me a while to remember what this was for but im 99% sure that it is a compensation for/check on a function that returns an empty/invalid CurrentBook object in some condition that is not supposed to be possible (as long as the other functions are used correctly)